Crash if a test reverts

medusa from `main` (8d1519ba19a14c3fdd8a5e314cdbd4762eb1f54f) seems to crash if a test reverts:

```solidity
contract Test {
  function fuzz_revert_always() public returns (bool){
    revert();
  }
}
```

To reproduce the crash:

```
$ medusa fuzz --target crash.sol
...
fuzz: elapsed: 0s, call: 0 (0/sec), seq/s: 0, resets/s: 0, cov: 0
Creating 10 workers ...

Fuzzer stopped, test results follow below ...
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xaf2a74]

goroutine 1 [running]:
github.com/crytic/medusa/fuzzing/executiontracer.(*ExecutionTrace).String(0xc0006b4040?)
	/home/g/Code/medusa/fuzzing/executiontracer/execution_trace.go:277 +0x14
github.com/crytic/medusa/fuzzing.(*PropertyTestCase).Message(0xc0003bd790)
	/home/g/Code/medusa/fuzzing/test_case_property.go:49 +0xbf
github.com/crytic/medusa/fuzzing.(*Fuzzer).printExitingResults(0xc0000cb440)
	/home/g/Code/medusa/fuzzing/fuzzer.go:697 +0x333
github.com/crytic/medusa/fuzzing.(*Fuzzer).Start(0xc0000cb440)
	/home/g/Code/medusa/fuzzing/fuzzer.go:597 +0x55a
github.com/crytic/medusa/cmd.cmdRunFuzz(0x13b26e0?, {0xc0000ccf80?, 0x2?, 0x2?})
	/home/g/Code/medusa/cmd/fuzz.go:125 +0x397
github.com/spf13/cobra.(*Command).execute(0x13b26e0, {0xc0000ccf60, 0x2, 0x2})
	/home/g/go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:916 +0x862
github.com/spf13/cobra.(*Command).ExecuteC(0x13b2ca0)
	/home/g/go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:1044 +0x3bc
github.com/spf13/cobra.(*Command).Execute(...)
	/home/g/go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:968
github.com/crytic/medusa/cmd.Execute(...)
	/home/g/Code/medusa/cmd/root.go:21
main.main()
	/home/g/Code/medusa/main.go:11 +0x33
```

--- a/fuzzing/calls/call_message_abi_values.go
+++ b/fuzzing/calls/call_message_abi_values.go
@@ -8,6 +8,10 @@ import (
 	"github.com/ethereum/go-ethereum/common"
 )
 
+// UNRESOLVED_METHOD describes the constant that is returned when we are unable to resolve an abi.Method at runtime.
+// This primarily may occur when replaying a corpus and the ABI of a contract has changed between runs.
+const UNRESOLVED_METHOD = "UNRESOLVED_METHOD"
+
 // CallMessageDataAbiValues describes a CallMessage Data field which is represented by ABI input argument values.
 // This is represented at runtime by an abi.Method and its input values.
 // Note: The data may be serialized. When deserializing, the Resolve method must be called to resolve the abi.Method
@@ -87,9 +91,10 @@ func (d *CallMessageDataAbiValues) Resolve(contractAbi abi.ABI) error {
 // Pack packs all the ABI argument InputValues into call data for the relevant Method it targets. If this was
 // deserialized, Resolve must be called first to resolve necessary runtime data (such as the Method).
 func (d *CallMessageDataAbiValues) Pack() ([]byte, error) {
-	// We must have set an ABI method at runtime to serialize this.
+	// If we do not have an ABI method at runtime to serialize this, we will return a []byte of UNRESOLVED_METHOD
+	// This should only happen when the corpus is being replayed and the ABI of a contract has changed between runs
 	if d.Method == nil {
-		return nil, fmt.Errorf("ABI call data packing failed, method definition was not set at runtime")
+		return []byte(UNRESOLVED_METHOD), nil
 	}
 
 	// If our ABI method was not set, we can't serialize our data.

--- a/fuzzing/calls/call_message_abi_values.go
+++ b/fuzzing/calls/call_message_abi_values.go
@@ -8,10 +8,6 @@ import (
 	"github.com/ethereum/go-ethereum/common"
 )
 
-// UNRESOLVED_METHOD describes the constant that is returned when we are unable to resolve an abi.Method at runtime.
-// This primarily may occur when replaying a corpus and the ABI of a contract has changed between runs.
-const UNRESOLVED_METHOD = "UNRESOLVED_METHOD"
-
 // CallMessageDataAbiValues describes a CallMessage Data field which is represented by ABI input argument values.
 // This is represented at runtime by an abi.Method and its input values.
 // Note: The data may be serialized. When deserializing, the Resolve method must be called to resolve the abi.Method
@@ -91,10 +87,10 @@ func (d *CallMessageDataAbiValues) Resolve(contractAbi abi.ABI) error {
 // Pack packs all the ABI argument InputValues into call data for the relevant Method it targets. If this was
 // deserialized, Resolve must be called first to resolve necessary runtime data (such as the Method).
 func (d *CallMessageDataAbiValues) Pack() ([]byte, error) {
-	// If we do not have an ABI method at runtime to serialize this, we will return a []byte of UNRESOLVED_METHOD
-	// This should only happen when the corpus is being replayed and the ABI of a contract has changed between runs
+	// If we do not have an ABI method at runtime to serialize this, we will return an error.
+	// This may happen when the corpus is being replayed and the ABI of a contract has changed between runs.
 	if d.Method == nil {
-		return []byte(UNRESOLVED_METHOD), nil
+		return nil, fmt.Errorf("ABI call data packing failed, method definition was not set at runtime")
 	}
 
 	// If our ABI method was not set, we can't serialize our data.

--- a/fuzzing/calls/call_sequence.go
+++ b/fuzzing/calls/call_sequence.go
@@ -94,8 +94,26 @@ func (cs CallSequence) Hash() (common.Hash, error) {
 			return common.Hash{}, err
 		}
 
-		// Hash the call message
-		_, err = hashProvider.Write(utils.MessageToTransaction(cse.Call).Hash().Bytes())
+		// Try to pack the call message and obtain a hash for it.
+		// This may panic if the ABI changed and the ABI method/function targeted does not resolve or the call
+		// could otherwise not be packed/serialized. If it does, we use fixed hash data instead.
+		var messageHashData []byte
+		func() {
+			// If the below operations to obtain a message/call hash fail, we instead substitute it with hardcoded
+			// hash data.
+			defer func() {
+				if r := recover(); r != nil {
+					messageHashData = common.Hash{}.Bytes()
+				}
+			}()
+
+			// Try to obtain a hash for the message/call. If this fails, we will replace it in the deferred panic
+			// recovery.
+			messageHashData = utils.MessageToTransaction(cse.Call).Hash().Bytes()
+		}()
+
+		// Hash the message hash data.
+		_, err = hashProvider.Write(messageHashData)
 		if err != nil {
 			return common.Hash{}, err
 		}

--- a/fuzzing/test_case_property.go
+++ b/fuzzing/test_case_property.go
@@ -47,7 +47,7 @@ func (t *PropertyTestCase) Message() string {
 		// If an execution trace is attached then add it to the message
 		if t.propertyTestTrace != nil {
 			// TODO: Improve formatting in logging PR
-			msg = fmt.Sprintf("%s\n Property test execution trace:\n%s", msg, t.propertyTestTrace.String())
+			msg += fmt.Sprintf("\nProperty test execution trace:\n%s", t.propertyTestTrace.String())
 		}
 		return msg
 	}

--- a/fuzzing/test_case_property.go
+++ b/fuzzing/test_case_property.go
@@ -38,16 +38,18 @@ func (t *PropertyTestCase) Name() string {
 func (t *PropertyTestCase) Message() string {
 	// If the test failed, return a failure message.
 	if t.Status() == TestCaseStatusFailed {
-		return fmt.Sprintf(
-			"Test \"%s.%s\" failed after the following call sequence:\n%s\n"+
-				"Property test \"%s.%s\" execution:\n%s",
+		msg := fmt.Sprintf(
+			"Property test \"%s.%s\" failed after the following call sequence:\n%s",
 			t.targetContract.Name(),
 			t.targetMethod.Sig,
 			t.CallSequence().String(),
-			t.targetContract.Name(),
-			t.targetMethod.Sig,
-			t.propertyTestTrace.String(),
 		)
+		// If an execution trace is attached then add it to the message
+		if t.propertyTestTrace != nil {
+			// TODO: Improve formatting in logging PR
+			msg = fmt.Sprintf("%s\n Property test execution trace:\n%s", msg, t.propertyTestTrace.String())
+		}
+		return msg
 	}
 	return ""
 }

